### PR TITLE
Update dependencies and copyright info

### PR DESCRIPTION
**Pull request by sethrj**
I've gone through our CMake dependencies and updated the documentation in the README. I've also updated the contribution guidelines to reference the GitHub copyright info, since (in an unrelated project) I was recently asked to sign a dumb retroactive CLA (which ORNL legal rejected).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,21 +6,27 @@ requirements of LHC-HL upgrade.
...
+- [CUDA](https://developer.nvidia.com/cuda-toolkit): on-device computation
+- an MPI implementation (such as [Open MPI](https://www.open-mpi.org)): shared-memory parallelism
+- [https://root.cern](ROOT): I/O
+- [https://github.com/nlohmann/json](nljson): simple text-based I/O for
+  diagnostics and program setup
+- [https://gitlab.cern.ch/VecGeom/VecGeom](VecGeom): on-device navigation of GDML-defined detector geometry
+- [https://geant4.web.cern.ch/support/download](Geant4): preprocessing physics data for a problem input
+- [https://geant4.web.cern.ch/support/download](G4EMLOW): EM physics model data
+- [http://hepmc.web.cern.ch/hepmc/](HepMC3): Event input
+- [http://swig.org](SWIG): limited set of Python wrappers for analyzing input
```
Comment (paulromano) [inline README.md:22]: The markup is reversed here on all the links (should be text in brackets, links in parentheses)

**Review by paulromano — CHANGES_REQUESTED** [review]
Thanks for this update! Nice to be clear about contributing guidelines (notably the lack of need for CLA)

File: README.md (hunk 1 above)
Comment (sethrj) [reply README.md:22]: 🤦 thanks, will fix... interestingly it looks like it still renders correctly on github

**Review by paulromano — APPROVED** [review]
Looks good now!

**Pull request merged**